src/engine/strat_engine/thinpool.rs

```diff
@@ -185,7 +239,78 @@ impl ThinPool {
             .map(|s| s.length)
             .sum();
 
-        Ok(data_dev_used + spare_total + meta_dev_total)
+        let mdv_total = self.mdv.segments().iter().map(|s| s.length).sum();
+
+        Ok(data_dev_used + spare_total + meta_dev_total + mdv_total)
+    }
+
+    pub fn get_mut_filesystem_by_uuid(&mut self,
+                                      uuid: &FilesystemUuid)
+                                      -> Option<&mut StratFilesystem> {
+        self.filesystems.get_mut_by_uuid(uuid)
+    }
+
+    pub fn get_mut_filesystem_by_name(&mut self, name: &str) -> Option<&mut StratFilesystem> {
+        self.filesystems.get_mut_by_name(name)
+    }
+
+    pub fn has_filesystems(&self) -> bool {
+        !self.filesystems.is_empty()
+    }
+
+    /// Create a filesystem within the thin pool.
```

Comment:
You should either have this check that the name is not already in the pool or explicitly state the precondition that this must be a new name.